pkg/dockerutil/dockerutil.go

```diff
@@ -0,0 +1,73 @@
+package dockerutil
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+
+	"github.com/docker/docker/pkg/jsonmessage"
+	"github.com/docker/docker/pkg/term"
+	"github.com/remind101/empire/pkg/image"
+)
+
+// FakePull writes a jsonmessage stream to w that looks like a Docker pull.
+func FakePull(img image.Image, w io.Writer) error {
+	messages := []jsonmessage.JSONMessage{
+		{Status: fmt.Sprintf("Pulling repository %s", img.Repository)},
+		{Status: fmt.Sprintf("Pulling image (%s) from %s", img.Tag, img.Repository), Progress: &jsonmessage.JSONProgress{}, ID: "345c7524bc96"},
+		{Status: fmt.Sprintf("Pulling image (%s) from %s, endpoint: https://registry-1.docker.io/v1/", img.Tag, img.Repository), Progress: &jsonmessage.JSONProgress{}, ID: "345c7524bc96"},
+		{Status: "Pulling dependent layers", Progress: &jsonmessage.JSONProgress{}, ID: "345c7524bc96"},
+		{Status: "Download complete", Progress: &jsonmessage.JSONProgress{}, ID: "a1dd7097a8e8"},
+		{Status: fmt.Sprintf("Status: Image is up to date for %s", img)},
+	}
+
+	enc := json.NewEncoder(w)
+
+	for _, m := range messages {
+		if err := enc.Encode(&m); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// DecodeJSONMessageStream wraps an io.Writer to decode a jsonmessage stream into
+// plain text. Bytes written to w represent the decoded plain text stream.
+func DecodeJSONMessageStream(w io.Writer) *DecodedJSONMessageWriter {
```

Comment:
Also refactored out how the jsonmessage stream is docoded, so you can just wrap an io.Writer to decode the stream and write plain text.